### PR TITLE
strict iso8601 dates

### PR DESCRIPTION
**Pull request by lsh-0**
* utils.json_dumps now formats dates with a terminating 'Z' for utc.

### CODE REVIEW

File: src/publisher/tests/test_utils.py [hunk 1]
```diff
@@ -14,6 +14,12 @@ def setUp(self):
     def tearDown(self):
         pass
 
+    def test_json_dumps_rfc3339(self):
+        dt = datetime(year=2001, month=1, day=1, hour=23, minute=59, second=59, microsecond=123, tzinfo=pytz.utc)
```
Comment (giorgiosironi) [inline src/publisher/tests/test_utils.py]: would test this also with a different `tzinfo` being specified

**Comment by giorgiosironi** [comment]
`api-raml` now has the validation for this, so I'll upgrade it on this branch and check the result.

**Comment by giorgiosironi** [comment]
Ah no, I can't because this is lax and not bot-lax-adaptor. This is an additional barrier against serving bad data to journal, but we need the change in formatting to take place in bot-lax-adaptor for the article JSON to be valid right?

File: src/publisher/tests/test_utils.py (hunk 1 above)
Comment (lsh-0) [inline src/publisher/tests/test_utils.py]: good call

**Comment by lsh-0** [comment]
> but we need the change in formatting to take place in bot-lax-adaptor for the article 
> JSON to be valid right?

I don't think so actually. The dates that are being handled are 'published', 'statusDate' and 'versionDate' - all of which are handled by lax. I'll double check the 'published' date but I'm pretty sure lax updates that formatting too

**Comment by giorgiosironi** [comment]
Ah, so they are not in the JSON that bot-lax-adaptor produces at all? What's inside those fields there, some placeholder?

**Comment by giorgiosironi** [comment]
Trying out https://github.com/elifesciences/bot-lax-adaptor/pull/51 as a PR so that we can exclude there are other dates lying around.

**Comment by lsh-0** [comment]
> What's inside those fields there, some placeholder?
yup.

'published' doesn't need a placeholder, it's always in the xml.
'statusDate' and 'versionDate' are only added as placeholders: https://github.com/elifesciences/bot-lax-adaptor/blob/develop/src/validate.py#L203

**Pull request merged**